frontend/projects/setup-wizard/src/app/pages/embassy/embassy.page.ts

```diff
@@ -129,7 +139,9 @@ export class EmbassyPage {
     try {
       await this.stateService.setupEmbassy(logicalname, password)
       if (!!this.stateService.recoverySource) {
-        await this.navCtrl.navigateForward(`/loading`)
+        await this.navCtrl.navigateForward(`/loading`, {
+          queryParams: { action: this.incomingAction },
```

Comment:
`{ action: this.route.snapshot.paramMap.get('action') }`

Also, what happens if the query param is not there. Is that possible?

Reply:
Not in reality. Can assert that it will always be there. Otherwise, yes would have to handle the case of non-existence.